packages/backend-plugin-api/src/services/definitions/lifecycleServiceRef.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createServiceRef } from '../system/types';
+
+/**
+ * @public
+ **/
+export type ShutdownHookOptions = {
+  fn: () => Promise<void>;
```

Comment:
Maybe `void | Promise<void>`? What's our policy on being helpful in these instances :)

Reply:
I wonder if it'll be useful in these to be able to add an `after?: Array<ServiceRef<unknown>>` to ensure ordering. Maybe not in the first version of this.

Reply:
Make sense adding the void to the list. I like the after idea but it might be better/easy to add it once/if we need it.